ibm/data_source_ibm_is_endpoint_gateway_targets.go

```diff
@@ -0,0 +1,141 @@
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "CRN for this specific object",
+						},
+						isVPEResourceParent: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Parent for this specific object",
+						},
+						isVPEResourceName: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Display name in the requested language",
+						},
+						isVPEResourceType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Resource type of this offering.",
+						},
+						isVPEResourceEndpointType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Data endpoint type of this offering",
+						},
+						isVPEResourceFullQualifiedDomainNames: {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: "Fully qualified domain names",
+						},
+						isVPEResourceServiceLocation: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Service location of this offering",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMISEndpointGatewayTargetRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bmxSess, err := meta.(ClientSession).BluemixSession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	region := bmxSess.Config.Region
+	catalogManagementClient, err := meta.(ClientSession).CatalogManagementV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getCatalogOptions := &catalogmanagementv1.SearchObjectsOptions{}
+	// query := "kind%3Avpe+AND+svc+AND+parent_id%3Aus-south"
+	query := fmt.Sprintf("kind:vpe AND svc AND parent_id:%s", region)
+	getCatalogOptions.Query = &query
+	digest := false
+	getCatalogOptions.Digest = &digest
+	catalog, response, err := catalogManagementClient.SearchObjectsWithContext(context, getCatalogOptions)
+	if err != nil {
+		log.Printf("[DEBUG] GetCatalogWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+	if catalog != nil && *catalog.ResourceCount > 0 && catalog.Resources != nil {
+		resourceInfo := make([]map[string]interface{}, 0)
+		for _, res := range catalog.Resources {
+			l := map[string]interface{}{}
+			if res.ParentID != nil {
+				l[isVPEResourceParent] = *res.ParentID
+			}
+			l[isVPEResourceName] = "provider_cloud_service"
+			if res.Label != nil {
+				l[isVPEResourceType] = *res.Label
+			}
+			region := ""
+			data := res.Data
+			if data != nil {
+				if serviceCrn, ok := data["service_crn"].(string); ok {
+					if serviceCrn != "" {
+						l[isVPEResourceCRN] = serviceCrn
+						crnFs := strings.Split(serviceCrn, ":")
+						if len(crnFs) > 5 {
+							region = crnFs[5]
+						}
+						l[isVPEResourceServiceLocation] = region
+					}
+				}
+				l[isVPEResourceEndpointType] = data["endpoint_type"]
```

Comment:
check nil conditions for data["endpoint_type"], data["fully_qualified_domain_names"]